src/main/java/com/rambots4571/rampage/util/MotorController.java

```diff
@@ -0,0 +1,118 @@
+package com.rambots4571.rampage.util;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+
+import com.rambots4571.rampage.tools.PIDTuner;
+
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.util.sendable.Sendable;
+import edu.wpi.first.util.sendable.SendableBuilder;
+
+
+public class MotorController implements Sendable {
+  private final WPI_TalonFX motor;
+  private final PIDTuner tuner;
+
+  private int kPIDLoopIdx;
+  private final int timeoutMs = 10;
+
+  private double targetRPM, toleranceRPM;
+
+  public MotorController(WPI_TalonFX motor) {
+    this.motor = motor;
+    this.tuner = new PIDTuner();
+    this.tuner.setUpdater(this::updatePID);
+    this.kPIDLoopIdx = 0;
+  }
+
+  public void setkPIDLoopIdx(int kPIDLoopIdx) {
+    this.kPIDLoopIdx = kPIDLoopIdx;
+  }
+
+  public void setPIDF(double kP, double kI, double kD, double kF) {
+    this.tuner.setPIDF(kP, kI, kD, kF);
+    setMotorPIDF(kP, kI, kD, kF);
+  }
+
+  private void setMotorPIDF(double kP, double kI, double kD, double kF) {
+    motor.config_kP(kPIDLoopIdx, kP, timeoutMs);
+    motor.config_kI(kPIDLoopIdx, kI, timeoutMs);
+    motor.config_kD(kPIDLoopIdx, kD, timeoutMs);
+    motor.config_kF(kPIDLoopIdx, kF, timeoutMs);
+  }
+
+  /** This syncs up the tuner PIDF values with the motor. This should be run frequently */
+  public void updatePID(PIDTuner tuner) {
+    setMotorPIDF(tuner.getkP(), tuner.getkI(), tuner.getkD(), tuner.getkF());
+  }
+
+  public PIDTuner getTuner() {
+    return this.tuner;
+  }
+
+  /**
+   * How much RPM can it be off by
+   *
+   * @param rpm
+   */
+  public void setTolerance(double rpm) {
+    toleranceRPM = rpm;
+    motor.configAllowableClosedloopError(kPIDLoopIdx, convertRPMToRaw(rpm));
+  }
+
+  public double getRawSpeed() {
+    return motor.getSelectedSensorVelocity(kPIDLoopIdx);
+  }
+
+  public double convertRawToRPM(double ticksPer100ms) {
```

Comment:
these `convertRawToRPM()` and `convertRPMToRaw()` methods are still for our shooter i think it would be best to get rid of them

Reply:
Really? I feel like theres no harm in leaving them just because if we ever need to use rpm in whatever case we always have these methods (unless theres another reason). Also what units did you use to measure elevator speed?

Reply:
those two methods are used to check if the motor controller is on setpoint, and those methods are hard coded right now to give conversions if there is a 1:1 ratio in RPM but realisitically elevators are never 1:1 ratio and it wouldnt make sense to have that if its being used for something with a different gear. And from personal experience if this is just going to be used for an elevator which will probably have 1 motor and a follower motor these functionalities can be inside the subsystem class